File: URIS2023-AgilIS-Tim10-main/URIS2023_TempateProject-main/src/app/models/rag.ts
export class Rag{
    ragId: string
    ragValue: string
    backlogItemId: string
}


export class RagCreation {
    ragValue: string
    backlogItemId: string
 
    constructor(ragValue,backlogItemId ){
     this.ragValue=ragValue;
     this.backlogItemId =backlogItemId
    }
 }
 
 export class RagUpdate {
    ragId: string
     ragValue: string
 
     constructor(ragId, ragValue ){
       this.ragId =ragId;
       this.ragValue=ragValue;
        }

    }

File: URIS2023-AgilIS-Tim10-main/Rag_MicroService/Rag_MicroService/Models/Entities/Rag.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace Rag_MicroService.Models.Entities
{
    [Index(nameof(BacklogItemId), IsUnique = true)]
    public class Rag
    {
        [Key]
        public Guid RagId { get; set; }

        public string RagValue { get; set; }

        
        public Guid BacklogItemId { get; set; }

    }
}
